### ISSUE TITLE
Create corewars-parser-macro for inline warriors

### ISSUE THREAD

**Issue by ian-h-chamberlain** [corewa-rs/corewars#56]
[This blog](https://blog.m-ou.se/writing-python-inside-rust-1/) explains a way to write a proc-macro crate for compile-time syntax checking of the warrior. This would be nice to have for testing so that warriors could be written inline instead of as string literals.

Gated by https://github.com/rust-lang/rust/issues/54725 to use on stable (needs span information to preserve whitespace in the proc-macro).
